File: cmd/experimental/update.go
package experimental

import (
	"cast/util"
	"os"
	"os/exec"

	"github.com/darkmatterorg/orbit/utils"
	"github.com/spf13/cobra"
)

var UpdateCmd = &cobra.Command{
	Use:   "update",
	GroupID: "experimental",
	Short: "Update system",
	Run: func(cmd *cobra.Command, args []string) {
		//TODO add support for other package managers

		if utils.IsCmdInstalled("pacman") {
			cmdToRun := exec.Command("sudo", "pacman", "-Syyu")

			cmdToRun.Stdout = os.Stdout
			cmdToRun.Stderr = os.Stderr

			if err := cmdToRun.Run(); err != nil {
				util.Logger.Fatal("Failed to update system", "err", err)
			}
		} else {
			util.Logger.Fatal("Could not find a supported package manager")
		}
	},
}

File: cmd/experimental/update.py
import typer
import shutil
import os

from utils.logger import info, notice, error, success
from utils.checkIfRoot import checkIfRoot
from rich.console import Console

app = typer.Typer()
console = Console()

@app.command(rich_help_panel="Experimentals")
def update():
  """
  Update system
  """
  #TODO Support for more package managers

  checkIfRoot()

  if shutil.which("pacman"):
    try:
      os.system("pacman -Syyu --noconfirm")
    except:
      error("Failed to update system")
      raise typer.Exit(code=1)
    else:
      success("System updated")
  else:
    notice("Could not find supported package manager")